scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -267,48 +284,50 @@ All our SCE objects have the same `rowData` columns (as we can see in the next c
 ```{r compare rowdata}
 # Use `purrr::map()` to quickly extract rowData column names for all SCEs
 purrr::map(sce_list,
-           # This syntax is a formula where the period is a stand-in for 
-           # each item in sce_list
-           ~colnames(rowData(.)))
+           \(x) colnames(rowData(x)))
 ```
 
 
 #### Ensuring that only shared genes are used
 
 The next step in ensuring SCE compatibility is to make sure they all contain the same genes, which are stored as the SCE object's row names (these names are also found the `rowData` slot's row names). 
-Here, those gene ids are unique Ensembl gene ids.
+Here, those gene ids are unique Ensembl gene ids .
 
 We can use some `purrr` magic to quickly find the set of shared genes among our samples:
 
-```{r shared genes}
-shared_genes <- sce_list %>%
+```{r shared genes, live = TRUE}
+# Define vector of shared genes
+shared_genes <- sce_list |>
   # get rownames (genes) for each SCE in sce_list
-  purrr::map(rownames) %>%
+  purrr::map(rownames) |>
   # reduce to the _intersection_ among lists
   purrr::reduce(intersect)
+```
 
-# We've now created a vector of genes that are present in all SCEs
+```{r print shared genes, live = TRUE}
+# Use head to look at the vector of shared genes:
 head(shared_genes)
 ```
 
 In this case, we happen to know that all SCE objects we're working with already contained the same genes. 
-We can quickly confirm that this is true by looking at the number of rows across SCE objects, and we'll see that they are all the same:
+We do a quick-and-dirty check for this by looking at the number of rows across SCE objects, and we'll see that they are all the same:
 
-```{r check shared genes}
+```{r check shared genes, live = TRUE}
 # The number of genes in an SCE corresponds to its number of rows:
-sce_list %>%
+sce_list |>
   purrr::map(nrow)
 ```
 
 So, for our data, we will not have to subset to shared genes since they are already shared!
  
 #### Ensuring matching columns in `colData`
 
-Finally, we'll need to have the same column names across all SCE `colData` tables, so let's look at all those column names:
+Finally, we'll need to have the same column names across all SCE `colData` tables, so let's look at all those column names.
+We can use similar syntax here to what we used to look at all the `rowData` column names.
 
-```{r compare coldata}
+```{r compare coldata, live = TRUE}
```

Comment:
If you take my earlier suggestion about making the earlier `purrr::map` for the rowData to be live, you could then make this not live.